Addition of Standard Effective Temperature (SET) as an Output Variable

RKStrand opened the pull request:
Pull request overview
---------------------
Based on discussion on unmethours.com, there was a request to add Standard Effective Temperature (SET) as an output variable.  This is calculated as part of the Pierce two-node thermal comfort model.  This work fulfills this request and it is now possible to select SET as an output variable in EnergyPlus.  This includes code changes, a unit test, documentation updates, and a modified test suite input file to show the use of this new output variable.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue #7174 

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes n/a
 - [x] Required documentation updates?

RKStrand committed:
Code additions to produce SET

This commit contains the code necessary to produce the SET variable for
the Pierce Two-Node Thermal Comfort model.  If uses wish to have SET
for their run, they simply need to select Pierce as an option in the
People statement and then add syntax to produce the Zone Thermal
Comfort Pierce Model Standard Effective Temperature output variable.

RKStrand committed:
Unit test, doc changes, and minor edits

This commit includes a new unit test for this work, documentation
changes (I/O Ref) to describe the new output variable, and some other
minor code cleanups.

RKStrand committed:
Merge branch 'develop' into 7174-AddSETOutput

RKStrand committed:
Input file changes to reflect new output variable

Addition of syntax to an input file to show the new output variable.

mjwitte commented on tst/EnergyPlus/unit/ThermalComfort.unit.cc:
```diff
@@ -948,3 +948,527 @@ TEST_F(EnergyPlusFixture, ThermalComfort_CalcIfSetPointMetWithCutoutTest)
+    OutputProcessor::TimeValue.allocate(2);
+    DataGlobals::DDOnlySimulation = true;
+    
+    ManageSimulation();
```
Would prefer a much more lightweight unit test. Seems you could just have zone, people, and schedules in the idf snippet and then just call the functions to process zone and people data and then CalcThermalComfortPierce.

mjwitte replied:
Moderately important. This should get your schedule in place, if done before processing the people:
```
    ScheduleManager::ProcessScheduleInput();
    ScheduleManager::ScheduleInputProcessed = true;
```

RKStrand replied:
@mjwitte Ok, it's now much more lightweight.  Doesn't call anything except the Pierce subroutine.  Sets every manually in the unit test now.  This should be up in GitHub as of now.

mjwitte commented on src/EnergyPlus/ThermalComfort.cc:
```diff
@@ -467,6 +467,12 @@ namespace ThermalComfort {
                                     "State",
                                     People(Loop).Name);
+                SetupOutputVariable("Zone Thermal Comfort Pierce Model Standard Effective Temperature",
+                                    OutputProcessor::Unit::None,
```
OutputProcessor::Unit::C ??

mjwitte commented on tst/EnergyPlus/unit/ThermalComfort.unit.cc:
```diff
@@ -948,3 +948,527 @@ TEST_F(EnergyPlusFixture, ThermalComfort_CalcIfSetPointMetWithCutoutTest)
+    
+    ASSERT_TRUE(process_idf(idf_objects));
+    
+    OutputProcessor::TimeValue.allocate(2);
```
And when you merge in develop, this line will fail. Simply delete it.

RKStrand replied:
Will delete it.

RKStrand committed:
Merge branch 'develop' into 7174-AddSETOutput

RKStrand committed:
Code revisions in response to review comments

Two modifications were requested during the review process.  One was a
simple correction of the units for this new variable.  The other was a
revision (rewriting) of the unit test to make it more efficient.

RKStrand committed:
Missing commit changes

For some reason, this did not commit properly the first time?

rraustad commented:
New report variable added but no changes to table reports.

rraustad committed:
Remove unit test clear_state, update IO Ref.

rraustad committed:
Merge branch 'develop' of https://github.com/NREL/EnergyPlus into 7174-AddSETOutput

rraustad commented:
- Final code and unit test review, removed clear_state access from unit test
 - Compiled and reviewed IO Ref., small edits, no new figures to add to CMake.
 - Ran sequential unit tests in release mode without failure
 - Ran example file, no warnings in err file. Same for annual run.
 - Verified new report variable added to existing example file (TermReheatSurfTC.idf)
 - Pulled develop and uploaded final changes
 - Plotted new report variable for design days

![image](https://user-images.githubusercontent.com/5778156/63877602-ee7d2600-c995-11e9-842c-0f0e1988fe35.png)

rraustad reviewed (approved):
Minor editorial mods to IO Ref, correct unit test, uploaded changes.

rraustad commented:
This is ready to merge once CI has a chance to process changes.

rraustad commented:
@RKStrand can you comment on the SET being that much above the summer design day zone temp in the figure above (e.g., occupants have on too many clothes, etc.)? The summer zone humidity ratio is actually lower than winter and zone temp is not that much different. No other code changes will be requested, just wondering if this is working as expected (i.e., if new issue should be opened).

RKStrand commented:
@rraustad I see what you are saying in the results, but I think this is actually what one would expect.  SET includes both radiative and convective aspects as well as a bunch of other things.  I would expect that SET would exhibit somewhat similar behavior to MAT but the surface temperatures in the space will dampen the response and make SET more closely related to outside air temperature (since that will in turn affect inside surface temperatures).  So, I think that what you are seeing is actually correct--SET is lower than MAT for the most part for winter (because MRT is lower as a result of the cold exterior conditions) and SET is generally higher than MAT for summer.  The "heavy" surfaces are damping the response of SET so when MAT changes rapidly like it does at the end of the work day in the summer data MAT can flip so that it is higher than SET because MAT in that situation is actually higher temporarily for that time.  Basically, if one were to graph MAT, MRT, and SET, I would guess that in general SET should track between these two for the most part.  Hopefully that makes sense, but if it doesn't, please let me know.  Overall, I think what you are seeing is the right trend and I don't think there is a problem here that needs to be logged as a defect.  Thanks for asking and for diligently checking results to see if there are additional problems!

rraustad commented:
Followup graph based on results discussion.

![image](https://user-images.githubusercontent.com/5778156/63888461-f6949000-c9ac-11e9-8125-3c4b73ffd9d3.png)

RKStrand committed:
Merge branch 'develop' into 7174-AddSETOutput

rraustad commented:
I can't think of a reason why this should not merge except for a complete CI assessment...

RKStrand commented:
@Myoldmopar This has met with approval from @rraustad but it sounds like there wasn't a complete CI result?  Does it need to be relaunched or something?

Myoldmopar commented:
The CI machine itself needed to be relaunched.  And it has been done.  CI results should be coming in soon.

rraustad commented:
Diff's from Coil Sizing report fixes #7435 are creeping in here. Merged that one into develop 17 hrs ago.

RKStrand committed:
Merge branch 'develop' into 7174-AddSETOutput

RKStrand committed:
Merge branch 'develop' into 7174-AddSETOutput

rraustad commented:
Well, I was just going to merge this anyway since I understood the diff's. But it can't hurt to have a fresh clean CI result.

RKStrand commented:
@rraustad Oops--sorry to get in the way.  Should have left it alone.  But as you pointed out, a fresh CI can't (shouldn't) hurt.

Myoldmopar commented:
We've had an issue with the CI box and had to take it down.  I'm working to get some extra support stood back up but this is likely to need to wait until after the weekend.

rraustad commented:
CI machines now match with expected results.
